// react/index.js
export class Component {} export class PureComponent {}
export const Fragment = Symbol.for('react.fragment');
export function createElement(type, props, ...children) { return { type, props: {...props, children}, $$typeof: Symbol.for('react.element') }; }
export function createContext(defaultValue) { return { Provider: class { constructor(v){this._value=v} }, Consumer: class { constructor(f){this._fn=f} }, _currentValue: defaultValue, _currentValue2: defaultValue }; }
export function createRef() { return { current: null }; }
export function createFactory(type) { return (props, ...children) => createElement(type, props, ...children); }
export function forwardRef(render) { return render; }
export function isValidElement(obj) { return obj && obj.$$typeof === Symbol.for('react.element'); }
export function memo(component) { return component; }
export function lazy(loader) { return class extends Component { componentDidMount() { loader().then(m => this.setState({ default: m.default || m })); } render() { return null; } }; }
export function useCallback(callback) { return callback; }
export function useContext(context) { return context._currentValue; }
export function useDebugValue() {}
export function useDeferredValue(value) { return value; }
export function useEffect(effect, deps) { /* noop */ }
export function useId() { return ':' + Math.random().toString(36).slice(2, 11) + ':'; }
export function useImperativeHandle(ref, createHandle, deps) { /* noop */ }
export function useInsertionEffect(destruct, deps) { /* noop */ }
export function useLayoutEffect(effect, deps) { /* noop */ }
export function useMemo(factory, deps) { return factory(); }
export function useReducer(reducer, initialArg, init) { return [initialArg, () => {}]; }
export function useRef(initialValue) { return { current: initialValue }; }
export function useState(initialState) { return [initialState, () => {}]; }
export function useSyncExternalStore(subscribe, getSnapshot, getServerSnapshot) { return getSnapshot(); }
export function useTransition() { return [() => {}, false]; }
export default { Component, PureComponent, Fragment, createElement, createContext, createRef, createFactory, forwardRef, isValidElement, memo, lazy, useCallback, useContext, useDebugValue, useDeferredValue, useEffect, useId, useImperativeHandle, useInsertionEffect, useLayoutEffect, useMemo, useReducer, useRef, useState, useSyncExternalStore, useTransition };
