// react/index.js
export class Component {} export class PureComponent {}
export const Fragment = Symbol("Fragment");
export function createElement() { return {}; }
export function createContext() { return { Provider: {}, Consumer: {} }; }
export function createRef() { return { current: null }; }
export function createFactory() { return () => {}; }
export function forwardRef(render) { return render; }
export function isValidElement() { return true; }
export function memo(component) { return component; }
export function lazy() { return Component; }
export function useCallback(f) { return f; }
export function useContext() { return null; }
export function useDebugValue() {}
export function useDeferredValue(v) { return v; }
export function useEffect() {}
export function useId() { return 'id'; }
export function useImperativeHandle() {}
export function useInsertionEffect() {}
export function useLayoutEffect() {}
export function useMemo(f) { return f(); }
export function useReducer() { return [null, () => {}]; }
export function useRef() { return { current: null }; }
export function useState(initial) { return [initial, () => {}]; }
export function useSyncExternalStore() { return null; }
export function useTransition() { return [() => {}, false]; }
export default { Component, PureComponent, Fragment, createElement, createContext, createRef, createFactory, forwardRef, isValidElement, memo, lazy, useCallback, useContext, useDebugValue, useDeferredValue, useEffect, useId, useImperativeHandle, useInsertionEffect, useLayoutEffect, useMemo, useReducer, useRef, useState, useSyncExternalStore, useTransition };
