// ink/index.js
export class Box { constructor(props) { this.props = props || {}; } }
export class Text { constructor(props, children) { this.props = props || {}; this.children = children; } }
export class Static { constructor(props) { this.props = props || {}; } }
export class Newline { constructor(props) { this.props = props || {}; } }
export class Spacer { constructor(props) { this.props = props || {}; } }
export function useApp() { return { stdout: { write: () => {} }, stderr: { write: () => {} }, stdin: { on: () => {} } }; }
export function useInput(handler, options) { /* noop */ }
export function useStdin() { return { on: () => {} }; }
export function useStdout() { return { write: () => {} }; }
export function useStderr() { return { write: () => {} }; }
export function render(node, options) { return { unmount: () => {}, waitUntilExit: () => Promise.resolve() }; }
export function measureElement() { return { width: 0, height: 0 }; }
export function rerender(node) { /* noop */ }
export default { Box, Text, Static, Newline, Spacer, useApp, useInput, useStdin, useStdout, useStderr, render, measureElement, rerender };
