// vscode/index.js
// VSCode API stub for CLI mode
const vscode = {
  env: { machineId: 'stub-machine-id', sessionId: 'stub-session' },
  extensions: { all: [], get: () => null },
  workspace: { workspaceFolders: null, rootPath: process.cwd() },
  window: { showInformationMessage: () => {}, showWarningMessage: () => {}, showErrorMessage: () => {} },
  commands: { registerCommand: () => {}, executeCommand: () => {} },
  languages: { createDiagnosticCollection: () => ({ clear: () => {}, set: () => {} }) },
  StatusBarAlignment: { Left: 1, Right: 2 },
  CommentMode: { Expanded: 1, Collapsed: 2 },
  DiagnosticSeverity: { Error: 0, Warning: 1, Information: 2, Hint: 3 },
  TreeItem: class { constructor(label) { this.label = label; this.iconPath = null; this.contextValue = null; } },
  TreeItemCollapsibleState: { None: 0, Collapsed: 1, Expanded: 2 },
  ViewColumn: { Active: 1, Beside: 2, One: 1, Two: 2, Three: 3 },
  ProgressLocation: { SourceControl: 1, Window: 10, Notification: 15 },
  TextEditorRevealType: { Default: 0, InCenter: 1, AtTop: 2 },
  ConfigurationTarget: { Global: 1, Workspace: 2, WorkspaceFolder: 3 },
  EndOfLine: { LF: 1, CRLF: 2 },
  Severity: { Ignore: 0, Info: 1, Warning: 2, Error: 3 },
  ColorThemeKind: { Light: 1, Dark: 2, HighContrast: 3 },
  InputBoxValidationSeverity: { Error: 1, Warning: 2, Information: 4 },
  QuickInputButtons: { Back: { iconPath: null } },
  ThemeIcon: { class { constructor(id) { this.id = id; } } },
  MarkdownString: class { constructor(value = '') { this.value = value; this.isTrusted = false; } },
  Range: class { constructor(start, end) { this.start = start; this.end = end; } },
  Position: class { constructor(line, character) { this.line = line; this.character = character; } },
  Selection: class extends Range { constructor(anchor, active) { super(anchor, active); this.anchor = anchor; this.active = active; } },
  Location: class { constructor(uri, rangeOrPos) { this.uri = uri; this.range = rangeOrPos; } },
  Uri: class { static parse(s) { return new Uri(s); } static file(s) { return new Uri(s); } static joinPath(...p) { return new Uri(p.join('/')); } constructor(s) { this.scheme = 'file'; this.path = s; this.fsPath = s; this.toString = () => s; } },
  DocumentHighlight: class { constructor(range, kind = 1) { this.range = range; this.kind = kind; } },
  DocumentLink: class { constructor(range, target) { this.range = range; this.target = target; } },
  CodeLens: class { constructor(range, command) { this.range = range; this.command = command; } },
  FoldingRange: class { constructor(start, end, kind) { this.start = start; this.end = end; this.kind = kind; } },
  SymbolKind: { File: 0, Module: 1, Namespace: 2, Package: 3, Class: 4, Method: 5, Property: 6, Field: 7, Constructor: 8, Enum: 9, Interface: 10, Function: 11, Variable: 12, Constant: 13, String: 14, Number: 15, Boolean: 16, Array: 17, Object: 18, Key: 19, Null: 20, EnumMember: 21, Struct: 22, Event: 23, Operator: 24, TypeParameter: 25 },
  CodeActionKind: { Empty: '', QuickFix: 'quickfix', Refactor: 'refactor', RefactorExtract: 'refactor.extract', RefactorInline: 'refactor.inline', RefactorRewrite: 'refactor.rewrite', Source: 'source', SourceOrganizeImports: 'source.organizeImports', SourceFixAll: 'source.fixAll' },
  InlayHintKind: { Type: 1, Parameter: 2 },
  InlineValueKind: { Variable: 1, Expression: 2, Range: 3 },
  SemanticTokenTypes: { enum: ['comment', 'keyword', 'string', 'number', 'regexp', 'operator', 'namespace', 'type', 'struct', 'class', 'interface', 'enum', 'typeParameter', 'parameter', 'variable', 'property', 'label', 'enumMember', 'function', 'method', 'macro', 'variable', 'parameter', 'type', 'namespace', 'property', 'enum', 'class', 'struct', 'interface', 'typeParameter', 'parameter', 'variable', 'function', 'method'] },
  TerminalExitReason: { Unknown: 0, Shutdown: 1, ProcessExited: 2, Integrated: 3 },
  TestMessage: class { constructor(message) { this.message = message; this.expected = undefined; this.actual = undefined; } },
  TestTag: class { constructor(id) { this.id = id; } },
  TestRunProfileKind: { Run: 1, Debug: 2, Coverage: 3 },
  DataTransferItem: class { constructor(value) { this.value = value; } },
  CancellationToken: { isCancellationRequested: false, onCancellationRequested: null },
  Disposable: class { constructor(fn) { this.dispose = fn; } dispose() {} },
  EventEmitter: class { constructor() { this.event = null; } fire() {} dispose() {} },
  Memento: class { get(k, d) { return d; } set(k, v) {} update(k, v) {} },
  FileSystem: class { stat() { return Promise.resolve({ type: 1, size: 0, mtime: new Date() }); } readFile() { return Promise.resolve(Buffer.from('')); } writeFile() { return Promise.resolve(); } readDirectory() { return Promise.resolve([]); } createDirectory() { return Promise.resolve(); } delete() { return Promise.resolve(); } rename() { return Promise.resolve(); } copy() { return Promise.resolve(); } },
};
module.exports = vscode;
