// marked/index.js
export class Lexer { constructor(opts) { this.tokens = []; this.options = opts || {}; } lex(src) { return this.tokens; } }
export class Parser { constructor(opts) { this.tokens = []; this.options = opts || {}; } parse(tokens) { return ''; } }
export class Renderer { constructor(opts) { this.options = opts || {}; } heading(text, level) { return text; } paragraph(text) { return text; } }
export class Tokenizer { constructor(opts) { this.options = opts || {}; } }
export class Slugger { slugify(src) { return src.toLowerCase().replace(/[^a-z0-9]+/g, '-'); } }
export const defaults = { async: false, breaks: false, extensions: null, gfm: true, hooks: null, pedantic: false, renderer: null, silent: false, tokenizer: null, walkTokens: null };
export function marked(src, opts) { return typeof src === 'string' ? src : ''; }
export function parse(src, opts) { return marked(src, opts); }
export function parseInline(src, opts) { return marked(src, opts); }
export function lexer(src, opts) { return []; }
export function parser(tokens, opts) { return ''; }
export function walkTokens(token, walk) { /* noop */ }
export default { marked, parse, parseInline, lexer, parser, walkTokens, Lexer, Parser, Renderer, Tokenizer, Slugger, defaults };
